.github/workflows/publish_lambda_layer.yaml

```diff
@@ -0,0 +1,99 @@
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+name: Publish Lambda Layer to staging
+
+on:
+  workflow_call:
+    inputs:
+      artifact-name:
+        description: 'Name of actions/upload-artifact result'
+        required: true
+        type: string
+      component-version:
+        description: 'Version of APM Python library in this release'
+        required: true
+        type: string
+      architecture:
+        description: 'Must be x86_64 or arm64'
+        required: true
+        type: string
+      runtimes:
+        description: 'Space-delimited list of compatible runtimes'
+        required: true
+        type: string
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  publish_layer:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        aws_region: 
+          - ap-northeast-1
+          - ap-northeast-2
+          - ap-south-1
+          - ap-southeast-1
+          - ap-southeast-2
+          - ca-central-1
+          - eu-central-1
+          - eu-north-1
+          - eu-west-1
+          - eu-west-2
+          - eu-west-3
+          - sa-east-1
+          - us-east-1
+          - us-east-2
+          - us-west-1
+          - us-west-2
+
+    steps:
+      - name: Construct Layer Name
+        shell: bash
+        run: |
+          LAYER_NAME=solarwinds-apm-python
+          
+          if [[ "${{ inputs.architecture }}" != "x86_64" ]] && [[ "${{ inputs.architecture }}" != "arm64" ]] ; then
+            echo "ERROR: architecture must be one of x86_64, arm64"
+            exit 1
+          else
+            LAYER_NAME=$LAYER_NAME-${{ inputs.architecture }}
+            echo "ARCH=${{ inputs.architecture }}" >> $GITHUB_ENV
+          fi
+          
+          COMPONENT_VERSION_UNDERSCORES=$(echo ${{ inputs.component-version }} | sed -r 's/\./_/g')
+          LAYER_NAME=$LAYER_NAME-$COMPONENT_VERSION_UNDERSCORES
+          echo "LAYER_NAME=$LAYER_NAME" >> $GITHUB_ENV
+          
+          echo GITHUB_ENV:
+          cat $GITHUB_ENV
+
+      - name: Download built layer
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.artifact-name }}
+
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.LAMBDA_PUBLISHER_STAGING_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.LAMBDA_PUBLISHER_STAGING_SECRET_ACCESS_KEY }}
+          aws-region: ${{ matrix.aws_region }}
+
+      - name: Publish Lambda Layer
+        run: |
+          LAYER_ARN=$(
```

Comment:
We're capturing the resulting LAYER_ARN (nice!) but not sure where it's used--and this means an error exit code from `aws lambda` command will probably not cause the workflow to fail.

Reply:
Ah yes good point! Removed the capture in [666370a](https://github.com/solarwinds/apm-python/pull/221/commits/666370a5e422bf1402774372ec397f9f5eba8fbc).

We could use it like Otel does to print out the ARN in the workflow https://github.com/open-telemetry/opentelemetry-lambda/blob/cc5a0b6b3c24f3e4dff7a0e960ee87df83ee416e/.github/workflows/layer-publish.yml#L101

`echo "::notice ::$LAYER_ARN component-version=${{ inputs.component-version }}"`

They haven't done a publish of any layer in 4+ months so there's no example logs available.